### ISSUE TITLE
False negative: move ref to data

### ISSUE THREAD

**Issue by DGRHicron** [abaplint/abaplint#3093]
The below code doesn't work on versions 7.4sp08 and below. When I entered version 7.02/7.4sp02/7.4sp08 in syntax.version rule, there are no errors. But there should be. Am I wrong?

```
REPORT zfoobar.

DATA moo TYPE i VALUE 2.
DATA ref TYPE REF TO data.
CREATE DATA ref TYPE i.

ref->* = moo.
```
